CorsixTH/Lua/cheats.lua

```diff
@@ -169,3 +169,50 @@ function Cheats:cheatDecreasePrices()
     end
   end
 end
+
+--! Cheat fax function
+function Cheats:cheatByFax()
+  local cheatWindow = self.hospital.world.ui:getWindow(UICheats)
+  self:announceCheat()
+  -- If a cheats window is open, make sure the UI is updated
+  if cheatWindow then
+    cheatWindow:updateCheatedStatus()
+  end
+end
+
+--[[Cheats operated through Faxes go here]]
+--! Cheat that disables/enables Roujin's challenge (spawn rate cheat)
+function Cheats:cheatRoujin()
+  -- Roujin's challenge cheat
+  local hosp = self.hopsital
+  local ui = hosp.world.ui
+  if not hosp.spawn_rate_cheat then
+    ui.adviser:say(_A.cheats.roujin_on_cheat)
+    hosp.spawn_rate_cheat = true
+    self:cheatByFax()
+  else
+    ui.adviser:say(_A.cheats.roujin_off_cheat)
+    -- Clear the current month's spawns to give the player a break
+    hosp.world.spawn_dates = {}
+    hosp.spawn_rate_cheat = nil
+  end
+end
+
+--! Cheat that disables/enables need to rest and set walking speed to maximum
+function Cheats:cheatNoRest()
+  local hosp = self.hospital
+  local ui = hosp.world.ui
+  if not hosp.no_rest_cheat then
+    for _, staff in ipairs(self.hospital.staff) do
+      if staff.attributes["fatigue"] then
+        staff:wake(staff.attributes["fatigue"])
+      end
```

Comment:
Here you compensate for the fatique, and in staff fatique update you skip updating.
Wouldn't it be simpler to perform this action there so you don't have it here, and fatique updates are more closely together?

Reply:
Also, simply assign `0.0` to the fatique rather than this complicated update, imho.

Reply:
This one I'd want to double check as originally I did set fatigue directly to 0 (technically should be 0.00).
However, the `wake` function does two things:
```lua
function Staff:wake(amount)
  self:changeAttribute("fatigue", -amount) -- This one changes the fatigue value
  self:updateDynamicInfo() -- This one updates the UI
end
```

I could go with the route of mimicking this function manually in cheats, but I thought in this case it would be better to use the proper function (wake). I value the view on moving this all to inside `staff.lua`, and did also try this -- but it generates needless updates to the fatigue value. It seemed easier to just set it once on cheat activation, and the cause any execution of the `tire` function inside `staff.lua` to just exit.
Though I'm happy to move on this if there is a good example.

Reply:
I think what you did is justified.